packages/embed/src/utils/create-iframe/get-form-event-handler.ts

```diff
@@ -0,0 +1,33 @@
+import { ActionableOptions } from '../../base'
+
+export const getFormReadyHandler = (embedId: string, options: ActionableOptions) => {
+  return getFormEventHandler('form-ready', embedId, options.onReady)
+}
+
+export const getFormQuestionChangedHandler = (embedId: string, options: ActionableOptions) => {
+  return getFormEventHandler('form-screen-changed', embedId, options.onQuestionChanged)
+}
+
+export const getFormSubmitHandler = (embedId: string, options: ActionableOptions) => {
+  const getEventPayload = (ev: any) => ({ responseId: ev?.data?.response_id })
+
+  return getFormEventHandler('form-submit', embedId, options.onSubmit, getEventPayload)
+}
+
+export function getFormEventHandler(
+  eventType: string,
+  embedId: string,
+  callback?: (ev: any) => void,
+  transformEventPayload: (ev: any) => any = () => undefined
+) {
+  return (event: any) => {
+    if (event.data.type !== eventType) {
+      return
+    }
+    if (event.data.embedId !== embedId) {
+      return
+    }
+    const transformedEvent = transformEventPayload(event)
+    callback?.(transformedEvent)
```

Comment:
I would prefer to be more explicit:

```
if (!callback) {
  return
}
```

Reply:
isn't this why the operator even exists? 🙂

Reply:
I think it kind of hides the functionality but that might be personal preference. Also, with my suggestion you would skip `transformEventPayload` when callback is not defined :)

Reply:
`transformEventPayload` falls back to `(ev) => undefined` when it's not defined, meaning that if we don't expose anything explicitly, it's not exposed automatically. The fallback syntax is not the best there, it's easy to overlook :)